### PR TITLE
Add ccoctl create key pair command for google cloud

### PR DESCRIPTION
**Pull request by akhil-rane**
- [x] Add command to create keypair for google cloud
- [x] Refactor key pair generation code and make it common for all cloud providers
- [x] Create a common constants file for pkg/cmd/provisioning package

x-ref: https://issues.redhat.com/browse/CCO-118

### CODE REVIEW

File: cmd/ccoctl/main.go [hunk 1]
```diff
@@ -17,6 +18,7 @@ func main() {
 
 	rootCmd.AddCommand(aws.NewAWSCmd())
 	rootCmd.AddCommand(ibmcloud.NewIBMCloudCmd())
+	rootCmd.AddCommand(gcp.NewGCPCmd())
```
Comment (joelddiaz) [inline cmd/ccoctl/main.go:21]: alphabetize
```suggestion
	rootCmd.AddCommand(gcp.NewGCPCmd())
	rootCmd.AddCommand(ibmcloud.NewIBMCloudCmd())
```

File: pkg/cmd/provisioning/aws/constants.go [hunk 2]
```diff
@@ -1,33 +0,0 @@
-package aws
```
Comment (joelddiaz) [inline pkg/cmd/provisioning/aws/constants.go:1]: for git history purposes, can this file show up as a rename to pkg/cmd/provisioning/constants.go ?

**Review by joelddiaz — COMMENTED** [review]
:+1: 
minor nits

File: cmd/ccoctl/main.go (hunk 1 above)
Comment (akhil-rane) [reply cmd/ccoctl/main.go]: fixed

File: pkg/cmd/provisioning/aws/constants.go (hunk 2 above)
Comment (akhil-rane) [reply pkg/cmd/provisioning/aws/constants.go:1]: I tried but as there are changes in that file git does not show it as rename

Comment (joelddiaz) [reply pkg/cmd/provisioning/aws/constants.go:1]: wierd b/c it seems to have worked for pkg/cmd/provisioning/aws/create_key_pair.go → pkg/cmd/provisioning/create_key_pair.go , oh well

File: pkg/cmd/provisioning/gcp/gcp.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	Region         string
+	Name           string
+	CredRequestDir string
+	DryRun         bool
+}
+
+// NewGCPCmd implements the "gcloud" subcommand for the credentials provisioning
+func NewGCPCmd() *cobra.Command {
+	createCmd := &cobra.Command{
+		Use:   "gcloud",
```
Comment (joelddiaz) [inline pkg/cmd/provisioning/gcp/gcp.go:21]: So are we finally going with `gcp` as the subcommand? (cc @dgoodwin )

File: pkg/cmd/provisioning/aws/constants.go (hunk 2 above)
Comment (akhil-rane) [reply pkg/cmd/provisioning/aws/constants.go:1]: true, no clue what is happening for constants.go

**Comment by akhil-rane** [comment]
@joelddiaz this good for another round of review. Changed sub-command name from gcloud to gcp

**Review by joelddiaz — COMMENTED** [review]
/lgtm

File: pkg/cmd/provisioning/constants.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	// NameTagKey is the key of the "Name" tag applied to the AWS resources created by ccoctl
+	NameTagKey = "Name"
+	// Generated identity provider files
+	OidcBucketFilename          = "01-oidc-bucket.json"
+	OidcConfigurationFilename   = "02-openid-configuration"
+	OidcKeysFilename            = "03-keys.json"
+	IAMIdentityProviderFilename = "04-iam-identity-provider.json"
+	// Generated role files
+	RoleFilenameFormat       = "05-%d-%s-role.json"
+	RolePolicyFilenameFormat = "06-%d-%s-policy.json"
```
Comment (abhinavdahiya) [inline pkg/cmd/provisioning/constants.go:32]: nit: no reason to keep aws specific constants in a global package. keep them private and close to the usage in aws implementation.

Comment (akhil-rane) [reply pkg/cmd/provisioning/constants.go:32]: These are the constants which are shared among different files in cloud provider packages. There are two options here
1) to create one constants file for entire provisioning package
2) to create separate constants file for each cloud provider

I prefer option (1) as it is easy to manage

Comment (joelddiaz) [reply pkg/cmd/provisioning/constants.go:32]: if these are truly global constants, then the var name `CcoctlAWSResourceTagKeyPrefix` shouldn't say `AWS` in the var name. Same for `OwnedCcoctlAWSResourceTagValue`.
And the comments for those two vars and `NameTagKey` shouldn't mention AWS.

Comment (akhil-rane) [reply pkg/cmd/provisioning/constants.go:32]: `CcoctlAWSResourceTagKeyPrefix` and `OwnedCcoctlAWSResourceTagValue` are constants specific to aws , but they are shared between different files in provisioning/aws package. I think I should move them somewhere inside the package.

**Comment by akhil-rane** [comment]
addressed PR feedback. PTAL

**Review by joelddiaz — COMMENTED** [review]
:+1: i'll let @abhinavdahiya give a look since he also had some feedback

**Comment by abhinavdahiya** [comment]
this is looking great! 
/lgtm

**Pull request merged**